### ISSUE TITLE
Aggressive Matching

### ISSUE THREAD

**Issue by davidagross** [caveljan/holocene-timestamp#2]
I haven't run the code in a debugger or anything but here are a few notes I've taken of numbers getting addverted that shouldn't (based on prefixes or suffixes that should eliminate the number from being a match):

- Source lines of code on GitHub source pages: (140 sloc) --> (140 [10140 HE] sloc)
  - (2231 sloc) does not get matched
- https://github.com/caveljan/Holocene-Epoch-Addverter: 
  - ("2017" can become "2017 [12017 HE]") becomes
  - ("2017 [12017 HE]" can become "2017 [12017 HE] [12017 HE]")

I've noticed a few more that I can't remember or don't have the MWE source to link to but I'll use this issue to add them as they come up.

Thanks for doing this work--I love the extension!

**Comment by ly3xqhl8g9** [comment]
Thank you for the reply and for using the extension.

Yes, alas, the regular expression implementation can make silly, forced matches. 

Maybe the forthcoming [lookbehind regexp](https://github.com/tc39/proposal-regexp-lookbehind) will help with weeding more of them out. However, I place my hope in being able to model one day a neural network that will have a 99,9% detection of what is an year string and what is not.

**Comment by davidagross** [comment]
Wow, how nice to see the closure of this issue.  I had turned off the extension back then for a while and am only now seeing that all of the issues I remarked on back then are totally addressed.  Hoorah for regular expressions!

Thanks for updating this, and I've already turned it back on for all sites.  I'll be sure to let you know if I run into anything odd if you're interested in that kind of feedback here.
